src/Composer/ComposerWidget.vala

```diff
@@ -490,7 +490,26 @@ public class Mail.ComposerWidget : Gtk.Grid {
     }
 
     private void on_discard () {
-        discarded ();
+        var discard_dialog = new Granite.MessageDialog.with_image_from_icon_name (
+            _("Delete This Draft?"), // choice of 'Delete' over 'Discard' for clarity
+            _("Are you sure you want to delete this draft permanently?"),
```

Comment:
Instead of repeating yourself here, this should offer some kind of further explanation. You might want to try answering additional questions like "Will my draft be deleted from the server too or just this computer?" and "Will I be able to undo this decision?"

Reply:
I'm not able to answer that first question because I don't actually know much about the Mail app (because I'm never able to login with my Gmail). I will do my best beyond that, though